File: Python/DataAndReportExportSample/DataAndReportExport/ControllerBase.py
#! /usr/bin/env python3
# pylint: disable= no-name-in-module
from restapisdk.configuration import Environment
from restapisdk.models.tspublic_rest_v_2_session_gettoken_request \
    import TspublicRestV2SessionGettokenRequest
from restapisdk.restapisdk_client import RestapisdkClient


class ControllerBase:
    def getToken(self, baseUrl, userName, password):
        client = RestapisdkClient(content_type='application/json',
                                  accept_language='application/json',
                                  access_token='',
                                  environment=Environment.PRODUCTION,
                                  base_url=baseUrl,
                                  skip_ssl_cert_verification=True, )
        body = TspublicRestV2SessionGettokenRequest()
        body.user_name = userName
        body.password = password
        session_controller = client.session
        response = session_controller.get_token(body)
        return response.token

    def getClient(self, baseUrl, userName, password):
        client = RestapisdkClient(
            content_type='application/json',
            accept_language='application/json',
            access_token=ControllerBase.getToken(
                self, baseUrl, userName, password),
            environment=Environment.PRODUCTION,
            base_url=baseUrl,
            skip_ssl_cert_verification=True)

        return client

File: DotNet/DataAndReportExportSample/ControllerBase.cs
using System;
using System.IO;
using System.Collections.Generic;
using System.Globalization;
using System.Threading.Tasks;

using RESTAPISDK.Standard;
using RESTAPISDK.Standard.Models;
using RESTAPISDK.Standard.Controllers;
using RESTAPISDK.Standard.Utilities;
using RESTAPISDK.Standard.Exceptions;

namespace DataAndReportExport
{
    class ControllerBase
    {

        public async Task<string> GetToken(string p_BaseURL, string p_UserName, string p_Password)
        {
            RESTAPISDKClient client = new RESTAPISDKClient.Builder()
                .AcceptLanguage("application/json")
                .ContentType("application/json")
                .BaseUrl(p_BaseURL)
                .HttpClientConfig(config => config.NumberOfRetries(0))
                .Build();
            SessionController sessionController = client.SessionController;

            var body = new TspublicRestV2SessionGettokenRequest();
            body.UserName = p_UserName;
            body.Password = p_Password;
            body.AccessLevel = GetTokenAccessLevelEnum.FULL;
            body.TokenExpiryDuration = "6000";

            SessionLoginResponse result = await sessionController.GetTokenAsync(body);

            return result.Token;
        }


        public RESTAPISDKClient GetClient(string p_BaseURL, string p_UserName, string p_Password)
        {
            var v_AccessToken = GetToken(p_BaseURL, p_UserName, p_Password).Result;

            RESTAPISDKClient client = new RESTAPISDK.Standard.RESTAPISDKClient.Builder()
            .AccessToken(v_AccessToken)
            .AcceptLanguage("application/json")
            .ContentType("application/json")
            .Environment(RESTAPISDK.Standard.Environment.Production)
            .BaseUrl(p_BaseURL)
            .HttpClientConfig(config => config.NumberOfRetries(0))
            .Build();

            return client;
        }
    }
}
